From da37e6ce2a8cc315845440488cd2f8dab0a17bfe Mon Sep 17 00:00:00 2001
From: "John C. Frickson" <jfrickson@nagios.com>
Date: Wed, 7 Oct 2015 14:06:58 -0500
Subject: [PATCH] Fix missing argument in open calls

Commit b9f2c08dfdf31430f972a48781fc115c4af13cfe
from https://github.com/NagiosEnterprises/nsca

Fix for issue http://tracker.nagios.org/view.php?id=286

Upstream-Status: Backport

Singed-off-by: John C. Frickson <jfrickson@nagios.com>
Signed-off-by: Chen Qi <Qi.Chen@windriver.com>
[MA: Context updated to allow application to older version]
Signed-off-by: Mark Asselstine <mark.asselstine@windriver.com>
---
 Changelog  | 4 ++++
 src/nsca.c | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/Changelog b/Changelog
index 3a7edd4..00ea11e 100644
--- a/Changelog
+++ b/Changelog
@@ -2,6 +2,10 @@
 NSCA Changelog
 **************
 
+2.9.x - xx/xx/xxxx
+------------------
+- Fix missing argument in open calls (Xavier Bachelot / John Frickson)
+
 2.9.1 - 01/27/2012
 ------------------
 - Applied patch to allow packets arriving with a future time stamp (Daniel Wittenberg)
diff --git a/src/nsca.c b/src/nsca.c
index 6b0e436..0e70b8b 100644
--- a/src/nsca.c
+++ b/src/nsca.c
@@ -477,7 +477,7 @@ static int read_config_file(char *filename){
                             int checkresult_test_fd=-1;
                             char *checkresult_test=NULL;
                             asprintf(&checkresult_test,"%s/nsca.test.%i",check_result_path,getpid());
-                            checkresult_test_fd=open(checkresult_test,O_WRONLY|O_CREAT);
+                            checkresult_test_fd=open(checkresult_test,O_WRONLY|O_CREAT,S_IWUSR);
                             if (checkresult_test_fd>0){
                                     unlink(checkresult_test);
                                     }
-- 
2.11.0

